src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -0,0 +1,424 @@
+            {
+                if (capacityNeeded > bufferCapacity)
+                {
+                    throw CreateSslException("OpenSsl::Encrypt outBufferPtr should be null");
+                }
+
+                IntPtr outBufferPtr = buffer;
+
+                retVal = BioRead(context.writeBioPtr, outBufferPtr, capacityNeeded);
+
+                if (retVal < 0)
+                {
+                    throw CreateSslException("OpenSsl::Encrypt failed");
+                }
+            }
+
+            return retVal;
+        }
+
+        internal static int Decrypt(IntPtr sslContextPtr, IntPtr outBufferPtr, int count)
+        {
+            SslContext context = Marshal.PtrToStructure<SslContext>(sslContextPtr);
+
+            var retVal = BioWrite(context.readBioPtr, outBufferPtr, count);
+
+            if (retVal == count)
+            {
+                retVal = libssl.SSL_read(context.sslPtr, outBufferPtr, retVal);
+
+                if (retVal > 0) count = retVal;
+            }
+
+            if (retVal != count)
+            {
+                throw CreateSslException("OpenSsl::Decrypt failed");
+            }
+            return retVal;
+        }
+
+        internal static IntPtr GetPeerCertificate(IntPtr sslContextPtr)
+        {
+            SslContext context = Marshal.PtrToStructure<SslContext>(sslContextPtr);
+            IntPtr sslPtr = context.sslPtr;
+            IntPtr certPtr = libssl.SSL_get_peer_certificate(sslPtr);
+            return certPtr;
+        }
+
+        internal static libssl.SSL_CIPHER GetConnectionInfo(IntPtr sslContextPtr)
+        {
+            SslContext context = Marshal.PtrToStructure<SslContext>(sslContextPtr);
+            IntPtr sslPtr = context.sslPtr;
+            IntPtr cipherPtr = libssl.SSL_get_current_cipher(sslPtr);
+            var cipher = new libssl.SSL_CIPHER();
+            if (IntPtr.Zero != cipherPtr)
+            {
+                cipher = Marshal.PtrToStructure<libssl.SSL_CIPHER>(cipherPtr);
+            }
+
+            return cipher;
+        }
+
+        internal static byte[] GetCertificateRawData(IntPtr certPtr)
+        {
+            byte[] buffer = null;
+            unsafe
+            {
+                byte* bufferPtr = null;
+                int len = libssl.i2d_X509(certPtr, ref bufferPtr);
+                if (len > 0)
+                {
+                    buffer = new byte[len];
+                    fixed (byte* pinnedBuffer = buffer)
+                    {
+                        bufferPtr = pinnedBuffer;   // TODO: creating temp var just in case
+                        len = libssl.i2d_X509(certPtr, ref bufferPtr);
+                    }
+                }
+                if (len <= 0)
+                {
+                    throw CreateSslException("Failed to get certificate raw data");
+                }
+            }
+
+            return buffer;
+        }
+
+        internal static void FreeCertificate(IntPtr certPtr)
```

Comment:
This isn't called, and can be deleted.  (Which is good, because I was going to have lots of questions otherwise)